python/ambassador/envoy/v2/v2ready.py

```diff
@@ -0,0 +1,153 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License
+
+from typing import List, TYPE_CHECKING
+
+from .v2listener import V2Listener
+
+if TYPE_CHECKING:
+    from . import V2Config # pragma: no cover
+
+
+class V2Ready(dict):
+
+    @classmethod
+    def generate(cls, config: 'V2Config') -> None:
+        # Inject the ready listener to the list of listeners if enabled
+        rport = config.ir.aconf.module_lookup('ambassador', 'ready_port', -1)
+        if rport <= 0:
+            config.ir.logger.info(f"V2Ready: ==== disabled")
+            return
+
+        rip = config.ir.aconf.module_lookup('ambassador', 'ready_ip', '127.0.0.1')
+        rlog = config.ir.aconf.module_lookup('ambassador', 'ready_log', True)
+
+        config.ir.logger.info(f"V2Ready: ==== listen on %s:%s" % (rip, rport))
+
+        typed_config = {
+            '@type': 'type.googleapis.com/envoy.config.filter.http.health_check.v2.HealthCheck',
+            'pass_through_mode': False,
+            'headers': [
+                {
+                    'name': ':path',
+                    'exact_match': '/ready'
+                }
+            ]
+        }
+
+        if rlog:
+            typed_config['access_log'] = cls.access_log(config)
+
+        ready_listener = {
+            'name': 'ambassador-listener-ready-%s-%s' % (rip, rport),
+            'address': {
+                'socket_address': {
+                    'address': rip,
+                    'port_value': rport,
+                    'protocol': 'TCP'
+                }
+            },
+            'filter_chains': [
+                {
+                    'filters': [
+                        {
+                            'name': 'envoy.filters.network.http_connection_manager',
+                            'typed_config': {
+                                '@type': 'type.googleapis.com/envoy.config.filter.network.http_connection_manager.v2.HttpConnectionManager',
+                                'stat_prefix': 'ready_http',
+                                'route_config': {
+                                    'name': 'local_route'
+                                },
+                                'http_filters': [
+                                    {
+                                        'name': 'envoy.filters.http.health_check',
+                                        'typed_config': typed_config
+                                    },
+                                    {
+                                        'name': 'envoy.filters.http.router'
+                                    }
```

Comment:
What's the point in including this?

Reply:
IIRC, that's needed right? (I'm not familiar with envoy config, I just reused something similar to config generated).
IIRC, I also tested without and it was not working. I can try again.